### PR TITLE
IPM fix for esp32s3 targets

### LINKED ISSUE #80890: IPM sample for esp32s3_devkitm remote build error
**Describe the bug**

Trying to build IPM sample for esp32s3_devkitm target is failing on remote stage.

```
CMake Error at cmake/modules/sysbuild_extensions.cmake:514 (message):
  CMake configure failed for Zephyr project: ipm_esp32_remote
```

**To Reproduce**

```
west build -b esp32s3_devkitm samples/drivers/ipm/ipm_esp32/ -p --sysbuild
```

**Expected behavior**

All necessary images should be built without an error.

**Impact**

Showstopper.

**Environment (please complete the following information):**

Upstream Zephyr + hal_espressif

### PR DESCRIPTION
**Pull request by marekmatej**
This PR fixes how the APPCPU is being built in IPM samples. C array encapsulation is removed and instead normal build procedure is used. Fixing the `sysbuild` builds as well.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/80890

### CODE REVIEW

File: soc/espressif/esp32s3/Kconfig.amp
Comment (uLipe) [inline soc/espressif/esp32s3/Kconfig.amp:11]: why it was decreased to 64KB?

File: soc/espressif/esp32s3/Kconfig.amp
Comment (uLipe) [inline soc/espressif/esp32s3/Kconfig.amp:23]: If this config is hidden you can remove the option prompt, otherwise it will became user selectable.

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:146]: Is it still a Todo?

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:160]: Typo: allready

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:172]: What this 0x20 mean?

**Review by uLipe — COMMENTED** [review]
Great to see this being sent.

I just left some initial commits.

File: soc/espressif/esp32s3/Kconfig.amp
Comment (marekmatej) [reply soc/espressif/esp32s3/Kconfig.amp:11]: It is the same setup as before due to a different approach. Now both IRAM and DRAM adds to allocate the space.

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (marekmatej) [reply soc/espressif/esp32s3/esp32s3-mp.c]: missed this

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (marekmatej) [reply soc/espressif/esp32s3/esp32s3-mp.c:168]: added explanation

File: samples/drivers/ipm/ipm_esp32/src/main.c
Comment (marekmatej) [reply samples/drivers/ipm/ipm_esp32/src/main.c:59]: Just a user experience.

File: dts/xtensa/espressif/esp32s3/esp32s3_common.dtsi
Comment (marekmatej) [reply dts/xtensa/espressif/esp32s3/esp32s3_common.dtsi:104]: same, reflects the updated layout from `memory.h`

File: soc/espressif/esp32s3/Kconfig
Comment (marekmatej) [reply soc/espressif/esp32s3/Kconfig]: fixed

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:57]: Just to avoid problems with cache coherence, it would not be good to invalidate D-Cache here before each time we move the data? Of course just in case we're moving data from the flash to the DRAM.

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:70]: Maybe same here, we can do even better, before loading the image, disable both I and D cache to avoid cache coherency issues.

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (uLipe) [inline soc/espressif/esp32s3/esp32s3-mp.c:144]: What about replace 1 to ESP32_APPCPU_INDEX or something in that sense?

File: soc/espressif/esp32s3/soc.c
Comment (uLipe) [inline soc/espressif/esp32s3/soc.c:179]: Why not use CONFIG_APPLICATION_PRIORITY instead of the plain 99?

**Review by uLipe — COMMENTED** [review]
I think we're almost there, I just left some extra questions

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (marekmatej) [reply soc/espressif/esp32s3/esp32s3-mp.c:57]: Since updating ld script for the APPCPU did not get the `STORE_PROHIBITED` exception anymore, and I'd rather not change this now.

File: soc/espressif/esp32s3/esp32s3-mp.c
Comment (marekmatej) [reply soc/espressif/esp32s3/esp32s3-mp.c:144]: we have 2 main cores 0 and 1, don't think this is necessary.

**Review by uLipe — DISMISSED** [review]

**Review by uLipe — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Comment by uLipe** [comment]
@mmahadevan108 any chance to put this into the 4.0? 

It is an important fix that will enable some samples for esp32 that are broken since HVM2 introduction.

Thank you :)

**Comment by mmahadevan108** [comment]
pinging @nordicjm for a review from HWV2 side as these samples were broken since the switch to HWV2.

**Review by wmrsouza — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@uLipe . This is a fairly big change. Is everything required for fixing the bug?
It might be easier to pull this into 4.0 if we can focus this PR to only include the changes needed to fix the bug and move the remaining to a separate PR that can be merged post 4.0 release.

**Comment by marekmatej** [comment]
> This is a fairly big change.

@mmahadevan108 you are right, and the reason it is so hefty is it not only fixes the IPM build itself but also stuff related to HWMv2, eg using sysbuild for multi-image builds.

**Comment by mmahadevan108** [comment]
@uLipe @marekmatej thank you for clarifying. This PR needs approval from @kartben as he is the assignee before we can merge

**Comment by marekmatej** [comment]
Now that assignee has been updated, I guess we can proceed as is? @mmahadevan108

**Comment by uLipe** [comment]
I'd say the same thig

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**
